components/automate-ui/src/app/components/landing/landing.component.ts

```diff
@@ -44,9 +48,14 @@ export class LandingComponent implements OnInit {
    */
   @Input() routePerms: RoutePerms[] = [];
 
-  public authorizedChecker: AuthorizedChecker;
-  private authComponent: AuthorizedComponent;
-  private index = 0;
+  /**
+   * A callback function to perform some actions in the event that no landing page was found.
+   * If no callback is provided, the caller guarantees one will always be found.
+   * (That should be the case for sidebars because we will never attempt to
+   * materialize a sidebar unless at least one item is permissible.)
+   * Thus, this defaults to a no-op function.
+   */
+  @Input() onNotFound: () => void = () => { };
```

Comment:
Not nearly as impressive as it looks. It is just a standard variable: `<name>: <type> = <initial value>`, but the type is a function `() => void` so the initial value is too `() => { }`.